test/integration/component/test_host_maintenance.py

```diff
@@ -0,0 +1,325 @@
+            "advanced",
+            "advancedns",
+            "smoke",
+            "basic",
+            "eip",
+            "sg"],
+        required_hardware="true")
+    def test_02_cancel_host_maintenace_with_migration_jobs(self):
+        
+        listHost = Host.list(
+            self.apiclient,
+            type='Routing',
+            zoneid=self.zone.id,
+            podid=self.pod.id,
+        )
+        for host in listHost:
+            self.logger.debug('2 Hypervisor = {}'.format(host.id))
+            
+        if (len(listHost) != 2):
+            raise unittest.SkipTest("Cancel host maintenance when VMs are migrating can only be tested with 2 hosts");
+            return
+
+        
+        no_of_vms = self.NoOfVMsOnHost(listHost[0].id)
+        
+        no_of_vms = no_of_vms + self.NoOfVMsOnHost(listHost[0].id)
+                
+        if no_of_vms < 5:
+            self.logger.debug("Create VMs as there are not enough vms to check host maintenance")
+            no_vm_req = 5 - no_of_vms
+            if (no_vm_req > 0):
+                self.logger.debug("Creating vms = {}".format(no_vm_req))
+                self.vmlist = self.createVMs(listHost[0].id, no_vm_req)
+        
+        vm_migrating=False
+        
+        try:
+           
+            target_host_id = listHost[0].id
+            other_host_id = listHost[1].id
+            
+            cmd = prepareHostForMaintenance.prepareHostForMaintenanceCmd()
+            cmd.id = target_host_id
+            response = self.apiclient.prepareHostForMaintenance(cmd)
+            
+            self.logger.debug('Host with id {} is in prepareHostForMaintenance'.format(target_host_id))
+            
+            #as soon as VM is picked for migration its last hostid is updated to the new host 
+            # that is why VM shows up as migrating on the other host
+            vm_migrating = wait_until(1, 10, self.checkVmMigratingOnHost, other_host_id)
+            
+            cmd = cancelHostMaintenance.cancelHostMaintenanceCmd()
+            cmd.id = target_host_id
+            response = self.apiclient.cancelHostMaintenance(cmd)
+            
+            self.logger.debug('Host with id {} is in cancelHostMaintenance'.format(target_host_id) )
+            
+                 
+            target_host_id = listHost[1].id
+            other_host_id = listHost[0].id
+            
+            cmd = prepareHostForMaintenance.prepareHostForMaintenanceCmd()
+            cmd.id = target_host_id
+            response = self.apiclient.prepareHostForMaintenance(cmd)
+            
+            self.logger.debug('Host with id {} is in prepareHostForMaintenance'.format(target_host_id))
+            
+            vm_migrating = wait_until(1, 10, self.checkVmMigratingOnHost, other_host_id)
+            
+            cmd = cancelHostMaintenance.cancelHostMaintenanceCmd()
+            cmd.id = target_host_id
+            response = self.apiclient.cancelHostMaintenance(cmd)
```

Comment:
Lines 296-309 appears to repeat with the only the role of the target_host and other_host varying.  Consider extracting into a function that parameterizes these variances to reduce the length of and clarifies the test functions.